bundler/lib/bundler/cli/gem.rb

```diff
@@ -389,8 +389,8 @@ def bundler_dependency_version
     end
 
     def ensure_safe_gem_name(name, constant_array)
-      if name =~ /^\d/
-        Bundler.ui.error "Invalid gem name #{name} Please give a name which does not start with numbers."
+      if name =~ /(^\d)|[A-Z]/
+        Bundler.ui.error "Invalid gem name #{name} Please give a name which does not start with numbers nor include capital letters."
```

Comment:
`only capital letters`?

Reply:
@simi This regex matches when there's only one capital letter.